adoc/chapters/device_compiler.adoc

```diff
@@ -216,132 +216,31 @@ The restriction waiver in <<discarded-statement>> or
+  [code]#bool#, [code]#char#, [code]#signed char#, [code]#unsigned char#,
+  [code]#short int#, [code]#unsigned short int#, [code]#int#, [code]#unsigned
+  int#, [code]#long int#, [code]#unsigned long int#, [code]#long long int#,
+  [code]#unsigned long long int#, [code]#float# and [code]#double#);
 
+- The fixed width types defined by the {cpp} core language (e.g.,
+  [code]#int8_t#, [code]#int16_t#, [code]#int32_t#, [code]#int64_t#); and
 
-[[table.types.fundamental]]
-.Fundamental data types supported by SYCL
-[width="100%",options="header",separator="@",cols="65%,35%"]
-|====
-@ Fundamental data type @ Description
-a@
-[source]
-----
-bool
-----
-   a@ A conditional data type which can be either true or false. The value
-      true expands to the integer constant 1 and the value false expands to the
-      integer constant 0.
-
-a@
-[source]
-----
-char
-----
-   a@ A signed or unsigned 8-bit integer, as defined by the {cpp} core language
-
-a@
-[source]
-----
-signed char
-----
-   a@ A signed 8-bit integer, as defined by the {cpp} core language
-
-a@
-[source]
-----
-unsigned char
-----
-   a@ An unsigned 8-bit integer, as defined by the {cpp} core language
-
-a@
-[source]
-----
-short int
-----
-   a@ A signed integer of at least 16-bits, as defined by the {cpp} core language
-
-a@
-[source]
-----
-unsigned short int
-----
-   a@ An unsigned integer of at least 16-bits, as defined by the {cpp} core language
-
-a@
-[source]
-----
-int
-----
-   a@ A signed integer of at least 16-bits, as defined by the {cpp} core language
-
-a@
-[source]
-----
-unsigned int
-----
-   a@ An unsigned integer of at least 16-bits, as defined by the {cpp} core language
-
-a@
-[source]
-----
-long int
-----
-   a@ A signed integer of at least 32-bits, as defined by the {cpp} core language
-
-a@
-[source]
-----
-unsigned long int
-----
-   a@ An unsigned integer of at least 32-bits, as defined by the {cpp} core language
-
-a@
-[source]
-----
-long long int
-----
-   a@ An integer of at least 64-bits, as defined by the {cpp} core language
-
-a@
-[source]
-----
-unsigned long long int
-----
-   a@ An unsigned integer of at least 64-bits, as defined by the {cpp} core language
-
-a@
-[source]
-----
-float
-----
-   a@ A 32-bit floating-point. The float data type must conform to the IEEE 754
-      single precision storage format.
-
-a@
-[source]
-----
-double
-----
-   a@ A 64-bit floating-point. The double data type must conform to the IEEE 754
-      double precision storage format.  This type is only supported on devices
-      that have [code]#aspect::fp64#.
-
-|====
+- Any type aliases defined by the {cpp} core language (e.g.,
+  [code]#std::size_t#, [code]#std::ptrdiff_t#).
 
+All types which are available in device code must have the same size, alignment
+requirements, and representation on both host and device.
+A device compiler may rely on additional command-line options to provide this
+guarantee for specific combinations of host and device.
```

Comment:
I don't think the latter sentence is helpful. The intent is to state what is required of a conforming implementation. An implementation that doesn't meet the stated requirements is non-conforming regardless of how it was configured.

Reply:
I agree; I only left it there originally because there was a similar sentence there before.  I've removed it in 22334ab7a7def1244223901363c0d70d3d4ae271.